vrs_anvil/evidence.py

```diff
@@ -367,3 +164,54 @@ def fetch_by_vrs_ids(
     data = result.fetchall()
     conn.close()
     return data
+
+
+def get_vcf_row(
+    variant_id: str, vcf: VariantFile, index_path: str = None
+) -> VariantRecord:
+    """given a variant id and annotated VCF, get the associated VCF row
+
+    Args:
+        variant_id (str): VRS ID for the variant of interest
+        vcf (VariantFile): Pysam VariantFile
+        index_path (str, optional): Index used to speed up search for variant. Defaults to iterating through VCF.
+
+    Raises:
+        Exception: outputs if no index is found
+
+    Returns:
+        VariantRecord: A Pysam VariantRecord (VCF row)
+    """
+
+    assert "VRS_Allele_IDs" in vcf.header.info, (
+        "no VRS_Allele_IDs key in INFO found, "
+        "please ensure that this is an VRS annotated VCF"
+    )
+
+    # try to populate from Bash env variable
+    if not index_path:
+        assert (
+            "VRS_VCF_INDEX" in os.environ
+        ), "no genotype index specified, no index path was provided nor was a variable name VRS_VCF_INDEX found."
+        index_path = Path(os.environ.get("VRS_VCF_INDEX"))
+
+    if index_path:
+        # if index provided, use it to get VCF row
+        index_path = Path(index_path)
+
+        # find variant of interest
+        for _, chr, pos in fetch_by_vrs_ids([variant_id], index_path):
+            # TODO [ISSUE-103]: generalize VCF fixtures
+            for record in vcf.fetch(f"chr{chr}", pos - 1, pos):
+                if variant_id in record.info["VRS_Allele_IDs"]:
+                    return record
+    else:
+        # otherwise, iterate through VCF
+        for record in enumerate(vcf.fetch()):
+            print(
+                "no VCF index specified, iterating through VCF to locate variant of interest"
+            )
+            if variant_id in record.info["VRS_Allele_IDs"]:
+                return record
+
+    raise Exception(f"no VCF row found matching variant ID {variant_id}")
```

Comment:
I think we can do better than a base Exception here -- maybe a `KeyError` if nothing else